sunpy/map/sources/psp.py

```diff
@@ -20,6 +22,37 @@ class WISPRMap(GenericMap):
     * `WISPR Instrument Page <https://wispr.nrl.navy.mil//>`__
     * `Instrument Paper <https://doi.org/10.1007/s11214-014-0114-y>`__
     """
+    # Default colormap bounds, per processing level and per detector. The image
+    # brightness varies significantly through each encounter, but these are
+    # reasonable overall defaults. Clipping at the high end is especially
+    # important, or else the automatic upper bound will be set by the stars and
+    # debris streaks, leaving the coronal structures much too dim (especially
+    # for L3).
+    colormap_presets = {
+        2: {
+            'I': (2e-12, 3e-10),
+            'O': (0, 1e-10),
+            },
+        3: {
+            'I': (0, 1.545e-11),
+            'O': (0, .5e-11),
+            }
+        }
+
+    def __init__(self, data, header, **kwargs):
+        super().__init__(data, header, **kwargs)
+
+        try:
+            detector = self.detector[0]
+            level = self.processing_level
+            vmin, vmax = self.colormap_presets[level][detector]
+        except KeyError:
```

Comment:
What case is this try-except catching? Are there files that don't have the detector or level set?

Reply:
Yeah, it's a failsafe for if the detector or level isn't in the collection of presets. For this and your other comment, the official data products should only ever have `1` or `2` for the `detector` entry in the header (I'm not a SOC member that can *guarantee* this, but I don't think they'll unveil a secret third detector any time soon). The `level` entry can also be `L1` or `L2b`---this PR was mostly just copying over snippets from my own WISPR tools, and since I don't work with those levels I didn't have presets ready, but they could be prepared.

Outside the official data products, others on the team produce `LW` and `LT` files using other processing techniques, and they put `LW` in the `level` field, for instance. One could also imagine someone producing a composite from the two imagers and putting some other value in the `detector` field to indicate this (or deleting the field altogether).

So I think either `WISPRMap` should be able to gracefully handle unexpected values in these fields, or `is_datasource_for` should only return `True` if these fields contain expected values. Not sure which is preferable.